### PR TITLE
EZP-30973: Made siteaccess aware repository to be used as default

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30973](https://jira.ez.no/browse/EZP-30973)
| **Bug/Improvement**| no
| **New feature**    | yes
| **Target version** | `master`
| **BC breaks**      | yes/no
| **Tests pass**     | yes
| **Doc needed**     | yes

**TODO**:
- [x] Rebase after https://github.com/ezsystems/ezpublish-kernel/pull/2815 merge
- [x] Rebase after https://github.com/ezsystems/ezpublish-kernel/pull/2833 merge
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

**QA**:
- [x] Make sure that site access aware repository layer is used when for instance `ezpublish.api.service.content` service is injected 
- [x] Run all possible sanities

### CODE REVIEW

File: eZ/Publish/Core/Repository/ObjectStateService.php [hunk 1]
```diff
@@ -224,7 +224,7 @@ public function updateObjectStateGroup(APIObjectStateGroup $objectStateGroup, Ob
             throw $e;
         }
 
-        return $this->buildDomainObjectStateGroupObject($spiObjectStateGroup);
+        return $this->buildDomainObjectStateGroupObject($spiObjectStateGroup, $objectStateGroup->prioritizedLanguages);
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/ObjectStateService.php]: Please add `@property-read string[] $prioritizedLanguages` to `\eZ\Publish\Core\Repository\Values\ObjectState\ObjectStateGroup`

File: eZ/Publish/Core/Repository/ObjectStateService.php [hunk 2]
```diff
@@ -380,7 +380,7 @@ public function updateObjectState(APIObjectState $objectState, ObjectStateUpdate
             throw $e;
         }
 
-        return $this->buildDomainObjectStateObject($spiObjectState);
+        return $this->buildDomainObjectStateObject($spiObjectState, null, $objectState->prioritizedLanguages);
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/ObjectStateService.php]: Please add `@property-read string[] $prioritizedLanguages` to `\eZ\Publish\Core\Repository\Values\ObjectState\ObjectState`

File: eZ/Publish/Core/settings/repository.yml [hunk 3]
```diff
@@ -1,19 +1,19 @@
 services:
     # API Aliases
     ezpublish.api.repository:
-        alias: eZ\Publish\Core\Event\Repository
+        alias: ezpublish.siteaccessaware.repository
```
Comment (alongosz) [inline eZ/Publish/Core/settings/repository.yml]: using named services instead of FQCN is a step back IMHO

**Comment by adamwojs** [comment]
PR updated according to @alongosz  code review suggestions. Additionally:
* Replaced `[]` to `\eZ\Publish\API\Repository\Values\Content\Language::ALL` in tests
* Added BC note

**Pull request merged**